docs/concepts/what-is-a-failure.md

```diff
@@ -0,0 +1,134 @@
+---
+id: what-is-a-failure
+title: What is a Failure?
+sidebar_label: Failure
+description: A Failure is Temporal's representation of various types of errors that occur in the system.
+tags:
+  - term
+  - explanation
+---
+
+A Failure is Temporal's representation of various types of errors that occur in the system.
+
+There are different types of Failures, and each has a different type in the SDKs and different information in the protobuf messages (which are used to communicate with the Cluster and appear in [Event History](/concepts/what-is-an-event-history)).
+
+## Temporal Failure
+
+Most SDKs have a base class that the other Failures extend:
+
+- TS: [`TemporalFailure`](https://typescript.temporal.io/api/classes/client.TemporalFailure)
+- Java: [`TemporalFailure`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/failure/TemporalFailure.html)
+- Python: [`TemporalError`](https://python.temporal.io/temporalio.exceptions.TemporalError.html)
+
+The base [`Failure` proto message](https://github.com/temporalio/api/blob/e381e51864ec8f43a90750ef936705258b8f64b2/temporal/api/failure/v1/message.proto#L81-L114) has these fields:
+
+- `string message`
+- `string stack_trace`
+- `string source`: The SDK this Failure originated in (for example, `"TypeScriptSDK"`). In some SDKs, this field is used to rehydrate the stack trace into an exception object.
+- `Failure cause`: The `Failure` message of the cause of this Failure (if applicable).
+
+## Application Failure
+
+Workflow and Activity code use Application Failures to communicate application-specific failures that happen.
+This is the only type of Failure created and thrown by user code.
+
+- TS: [`ApplicationFailure`](https://typescript.temporal.io/api/classes/client.ApplicationFailure)
+- Java: [`ApplicationFailure`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/failure/ApplicationFailure.html)
+- Go: [`ApplicationError`](https://pkg.go.dev/go.temporal.io/sdk/internal#ApplicationError)
+- Python: [`ApplicationError`](https://python.temporal.io/temporalio.exceptions.ApplicationError.html)
+- Proto: [`ApplicationFailureInfo`](https://github.com/temporalio/api/blob/e381e51864ec8f43a90750ef936705258b8f64b2/temporal/api/failure/v1/message.proto#L37-L41) and [`Failure`](https://github.com/temporalio/api/blob/e381e51864ec8f43a90750ef936705258b8f64b2/temporal/api/failure/v1/message.proto#L81-L114)
+
+### Throw from Workflows
+
+In Workflows, if you throw (or return in Go) an error that is not an Application Failure, the Workflow Task will fail and be retried.
+If you throw an Application Failure, the Workflow Execution will fail, and may be retried according to its Retry Policy.
+
+### Throw from Activities
+
+In Activities, you can either throw an Application Failure or another Error to fail the Activity Task.
```

Comment:
Looks like the full term is Activity Task Execution. I'm trying to say that it may just fail the Activity try, not the chain:

![image](https://user-images.githubusercontent.com/251288/200710596-c032b1cf-2206-4340-901d-56e087c8bea6.png)

Reply:
How come a "Workflow Execution" is a single attempt/run that can be retried, but an "Activity Execution" represents all retries? And we call what is an "Activity Execution" a "Workflow Execution Chain". This is confusing. As an SAT analogy, we have "Workflow Execution Chain":"Workflow Execution"::"Activity Execution":"Activity Task Execution"?

Maybe we can't fix it now or anything, but our inconsistent use of "execution" and "task" in the documentation is probably gonna be hard to grasp.

Reply:
> we have "Workflow Execution Chain":"Workflow Execution"::"Activity Execution":"Activity Task Execution"

yes. agree it's inconsistent and probably confusing

Reply:
Workflow Execution chain needs to exist because of Workflow Execution Timeout and Workflow Run Timeout 
Besides we don't actually make Workflow Execution chain a core term -

Otherwise - it is pretty consistent.

Reply:
> Workflow Execution chain needs to exist because of Workflow Execution Timeout and Workflow Run Timeout

"Activity Execution" includes "schedule to close timeout" and "start to close timeout" which are the same thing essentially.

>  Besides we don't actually make Workflow Execution chain a core term

I saw it at https://docs.temporal.io/workflows#workflow-execution-chain, but otherwise what do we call the workflow equivalent of an "Activity Execution" (i.e. the wrapper around attempts of retries)?

> Otherwise - it is pretty consistent.

Not really. "X Execution includes retries" is true where X = Activity, but not X = Workflow.